spectrum-integration/function/job/spectrum-user-job.ts

```diff
@@ -32,87 +33,81 @@ export class SpectrumUserJob implements IJob {
+                        break;
                     }
+                }
 
-                    //Build KPA user Data and Check existing
-                    if (kpaUser == null) {
-                        kpaUser = new KPAUserModel();
-                        if (user.employeeStatus !== 'A') {
-                            // console.log(`Skip User because of Termination Date ${user.employeeId} ${user.terminationDate}`)
-                            status.skippedRecord++
-                            continue;
-                        }
-                    } else {
-                        if (!this.isEditUser) {
-                            // console.log(`Skip User because of Cannot Allow to edit ${user.employeeId}`)
-                            status.skippedRecord++
-                            continue;
-                        }
-
-                        if (user.employeeStatus !== 'A') {
-                            status.inactivatedRecord++
-                        }
+                //Build KPA user Data and Check existing
+                if (kpaUser == null) {
+                    kpaUser = new KPAUserModel();
+                    if (user.employeeStatus !== 'A') {
+                        status.skippedRecord++
+                        continue;
+                    }
+                } else {
+                    if (!this.isEditUser) {
+                        status.skippedRecord++
+                        continue;
                     }
 
-                    //Create Users 
-                    kpaUser.employeeNumber = `${companyCode.trim()}-${user.employeeCode.trim()}`;
-                    kpaUser.firstName = user.firstName
-                    kpaUser.lastName = user.lastName;
-                    kpaUser.username = kpaUser.employeeNumber;
-                    kpaUser.email = '';
-                    kpaUser.initialPassword = `KPAFlex2024!!`;
-                    kpaUser.role = this.defaultRole;
-                    kpaUser.title = user.title
-                    kpaUser.welcomeEmail = this.welcomeEmail
-                    kpaUser.resetPassword = this.resetPassword
-
-                    if (user.employeeStatus !== 'A' && kpaUser.terminationDate == null) {
-                        kpaUser.terminationDate = new Date().toDateString();
-                        console.log(`Need to Check ${kpaUser.terminationDate}`)
+                    if (user.employeeStatus !== 'A') {
+                        status.inactivatedRecord++
                     }
+                }
 
-                    //Add User To List
-                    kpaUsers.push(kpaUser);
-                    status.upsertRecord++;
+                //Create Users
+                kpaUser.employeeNumber = `${companyCode.trim()}-${user.employeeCode.trim()}`;
+                kpaUser.firstName = user.firstName
+                kpaUser.lastName = user.lastName;
+                kpaUser.username = kpaUser.employeeNumber;
```

Comment:
Re-adding this change from last week. It got removed when merging in Yogi's branch.

```suggestion
                kpaUser.username = `${companyCode.trim()}-${user.employeeCode.trim()}`;
```

Reply:
You were correct in pulling in the latest change. I had merged in Yogis branch earlier though but it didn't have the latest updates from `main` pulled in. So it reverted just that one change.

https://github.com/HazardScout/anchorock-kpa/commit/86761802c8b347f49c6efa4cb411c2375b7034bc